app/models/shipit/task.rb

```diff
@@ -231,6 +235,10 @@ def supports_rollback?
       false
     end
 
+    def previous_success
+      Task.previous_successful(id)
```

Comment:
Might want to use `self.class.previous...` here, so that `Deploy#previous_success` won't return raw `Task` instances.

Reply:
Ah, thank you for this. I am still a bit green on Ruby's class system and the specifics of its polymorphism.